dashboard/serializers_test.py

```diff
@@ -388,3 +396,52 @@ def test_course_enrollments_serialized_unique(self):
         serialized_program_user = UserProgramSearchSerializer.serialize(self.non_fa_program_enrollment)
         # Number of serialized enrollments should be unaffected
         assert len(serialized_program_user['enrollments']) == serialized_count_before_addition
+
+
+class UserProgramSemesterSerializerEdxTests(TestCase):
+    """
+    Test cases for the serialization of a user's semester enrollments for search results
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        with mute_signals(post_save):
+            profile = ProfileFactory.create()
+        cls.user = profile.user
+        cls.program = ProgramFactory.create()
+        courses = CourseFactory.create_batch(7, program=cls.program)
+        runs = list()
+        # Add course runs that have a valid, edX-parse-able course key
+        runs.append(CourseRunFactory.create(course=courses[0], edx_course_key='course-v1:MITx+course1+1T2016'))
+        runs.append(CourseRunFactory.create(course=courses[1], edx_course_key='course-v1:MITx+course3+3T2016'))
+        # Add course runs that have an invalid key; they will be serialized based on their start_date
+        fall_2016_dt = datetime.strptime('10/01/2016', "%m/%d/%Y")
```

Comment:
Why are we parsing the datetime this way? Also, you can pass the string to the ORM and it will parse it into a date or datetime depending on the field